cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -198,13 +211,38 @@ private List<CloudResourceStatus> deleteResources(List<CloudResource> resources,
         azureGenericResources.forEach(cr -> persistenceNotifier.notifyDeletion(cr, cloudContext));
 
         return findResources(resources, List.of(AZURE_DATABASE)).stream()
-                .map(r -> deleteDatabaseServerAndNotify(r, cloudContext, client, persistenceNotifier, force))
+                .map(r -> deleteDatabaseServerAndNotify(r, cloudContext, client, persistenceNotifier, force, keyVaultUrl, keyVaultResourceGroupName))
                 .collect(Collectors.toList());
     }
 
     private CloudResourceStatus deleteDatabaseServerAndNotify(
-            CloudResource cloudResource, CloudContext cloudContext, AzureClient client, PersistenceNotifier persistenceNotifier, boolean force) {
+            CloudResource cloudResource, CloudContext cloudContext, AzureClient client, PersistenceNotifier persistenceNotifier, boolean force,
+            String keyVaultUrl, String keyVaultResourceGroupName) {
         LOGGER.debug("Deleting postgres server {}", cloudResource.getReference());
+        if (keyVaultUrl != null) {
+            String dbPrincipalId = client.getAzure().genericResources().getById(cloudResource.getReference()).inner().identity().principalId();
+            String vaultName = client.getVaultName(keyVaultUrl);
+            if (vaultName != null) {
+                // Check for the existence of keyVault user has specified before removing Database access permissions from this keyVault.
+                if (!client.keyVaultExists(keyVaultResourceGroupName, vaultName)) {
+                    LOGGER.warn(String.format("Vault with name \"%s\" either does not exist/have been deleted or user does not have permissions to access it.",
+                            vaultName));
+                } else {
+                    String description = String.format("access to Key Vault \"%s\" in Resource Group \"%s\" for Service Principal having object ID \"%s\" " +
+                                    "associated with Database.", vaultName, keyVaultResourceGroupName, dbPrincipalId);
+                    retryService.testWith2SecDelayMax15Times(() -> {
+                        try {
+                            LOGGER.info("Removing {}.", description);
+                            client.removeKeyVaultAccessPolicyFromServicePrincipal(keyVaultResourceGroupName, vaultName, dbPrincipalId);
```

Comment:
I just noticed the method name typo: Should not it rather be `ForServicePrincipal` instead of `FromServicePrincipal`?